state/state.go

```diff
@@ -101,7 +126,21 @@ func (s *State) Bytes() []byte {
 
 // Mutate state variables to match block and validators
 // after running EndBlock
-func (s *State) SetBlockAndValidators(header *types.Header, blockPartsHeader types.PartSetHeader, prevValSet, nextValSet *types.ValidatorSet) {
+func (s *State) SetBlockAndValidators(header *types.Header, blockPartsHeader types.PartSetHeader, abciResponses *ABCIResponses) {
+
+	// copy the valset
```

Comment:
I'd rather want to see the reason for copying than the duplicating message of what the code below does. This is my personal opinion.